src/Diagnostics.DataProviders/DataProviders/GeoMasterDataProvider.cs

```diff
@@ -481,6 +492,72 @@ public Task<string> GetLinuxContainerLogs(string subscriptionId, string resource
             return HttpPost<string, string>(path);
         }
 
+        /// <summary>
+        /// Gets all the APP SETTINGS for the Web App that start with prefixes like WEBSITE_, FUNCTION_ etc, filtering out
+        /// the sensitive settings like connectionstrings, tokens, secrets, keys, content shares etc.
+        /// </summary>
+        /// <param name="subscriptionId">Subscription Id for the resource</param>
+        /// <param name="resourceGroupName">The resource group that the resource is part of </param>
+        /// <param name="name">Name of the resource</param>
+        /// <param name="slotName">slot name (if querying for a slot, defaults to production slot)</param>
+        /// <returns>A dictionary of AppSetting Keys and values</returns>
+        /// <example>
+        /// <code>
+        /// This sample shows how to call the <see cref="GetAppSettings"/> method in a detector
+        /// public async static Task<![CDATA[<Response>]]> Run(DataProviders dp, OperationContext<![CDATA[<App>]]> cxt, Response res)
+        /// {
+        ///     var subId = cxt.Resource.SubscriptionId;
+        ///     var rg = cxt.Resource.ResourceGroup;
+        ///     var name = cxt.Resource.Name;
+        ///     var slot = cxt.Resource.Slot;
+        ///
+        ///     byte[] zip = await dp.GeoMaster.GetLogFilesZip(subId, rg, name, slot);
```

Comment:
Possibly put this in try catch since if detector authors copy paste the code from documentation, it won't impact our SLA.